[wip] Use codegen'ed unboxing wrappers

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32376 [wip] direct to unboxed func ptr
* **#30370 [wip] Use codegen'ed unboxing wrappers**

Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

The `use_c10_dispatcher` setting in `native_functions.yaml` now has a new option 'with_codegenerated_unboxing_wrapper' which means we take the codegened unboxing wrapper from register_aten_ops.cpp and stuff it into c10. This new argument is the default, 'unboxed_only' is not the default anymore. For the ops that don't support boxed dispatch yet (i.e. ops taking TensorOptions arguments), we set them to 'unboxed_only' and they follow the old behavior of having register_aten_ops.cpp register the jit op.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

smessmer committed:
[wip] Use codegen'ed unboxing wrappers

Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit fa1e7be:
* **1/1** failures introduced in this PR
* **1/1** recognized as flaky
    * Re-run these jobs?



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=fa1e7bedd10691d4337723ce0461a7669b7e8a9b).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3975683) binary_linux_manywheel_2_7mu_cpu_devtoolset7_test (1/1)
**Step:** "Run in docker" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75702015) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75702015))
<details>
<summary>
<code>Dec 17 02:52:02 ProtocolError: ("Connection broken: error(104, 'Connection reset by peer')", error(104, 'Connection reset by peer'))</code>
</summary>

```
Dec 17 02:52:02   File "/opt/_internal/cpython-2.7.15-ucs4/lib/python2.7/site-packages/pip/_internal/download.py", line 212, in resp_read 
Dec 17 02:52:02     decode_content=False): 
Dec 17 02:52:02   File "/opt/_internal/cpython-2.7.15-ucs4/lib/python2.7/site-packages/pip/_vendor/urllib3/response.py", line 564, in stream 
Dec 17 02:52:02     data = self.read(amt=amt, decode_content=decode_content) 
Dec 17 02:52:02   File "/opt/_internal/cpython-2.7.15-ucs4/lib/python2.7/site-packages/pip/_vendor/urllib3/response.py", line 529, in read 
Dec 17 02:52:02     raise IncompleteRead(self._fp_bytes_read, self.length_remaining) 
Dec 17 02:52:02   File "/opt/_internal/cpython-2.7.15-ucs4/lib/python2.7/contextlib.py", line 35, in __exit__ 
Dec 17 02:52:02     self.gen.throw(type, value, traceback) 
Dec 17 02:52:02   File "/opt/_internal/cpython-2.7.15-ucs4/lib/python2.7/site-packages/pip/_vendor/urllib3/response.py", line 443, in _error_catcher 
Dec 17 02:52:02     raise ProtocolError("Connection broken: %r" % e, e) 
Dec 17 02:52:02 ProtocolError: ("Connection broken: error(104, 'Connection reset by peer')", error(104, 'Connection reset by peer')) 
Dec 17 02:52:02 ++ sleep 1 
Dec 17 02:52:03 ++ pip install -q future numpy protobuf six 
Dec 17 02:52:04 DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support 
Dec 17 02:52:15 ++ [[ manywheel == libtorch ]] 
Dec 17 02:52:15 ++ /builder/check_binary.sh 
Dec 17 02:52:15 + [[ manywheel == libtorch ]] 
Dec 17 02:52:15 + py_dot=2.7 
Dec 17 02:52:15 +++ which python 
Dec 17 02:52:15 ++ dirname /opt/python/cp27-cp27mu/bin/python 
Dec 17 02:52:15 + install_root=/opt/python/cp27-cp27mu/bin/../lib/python2.7/site-packages/torch/ 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

The `use_c10_dispatcher` setting in `native_functions.yaml` now has a new option 'with_codegenerated_unboxing_wrapper' which means we take the codegened unboxing wrapper from register_aten_ops.cpp and stuff it into c10. This new argument is the default, 'unboxed_only' is not the default anymore. For the ops that don't support boxed dispatch yet (i.e. ops taking TensorOptions arguments), we set them to 'unboxed_only' and they follow the old behavior of having register_aten_ops.cpp register the jit op.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

The `use_c10_dispatcher` setting in `native_functions.yaml` now has a new option 'with_codegenerated_unboxing_wrapper' which means we take the codegened unboxing wrapper from register_aten_ops.cpp and stuff it into c10. This new argument is the default, 'unboxed_only' is not the default anymore. For the ops that don't support boxed dispatch yet (i.e. ops taking TensorOptions arguments), we set them to 'unboxed_only' and they follow the old behavior of having register_aten_ops.cpp register the jit op.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]

smessmer committed:
Update on "[wip] Use codegen'ed unboxing wrappers"


Not all ops support the templated unboxing wrappers yet. For the ones that don't,
let's use the codegen'ed unboxing wrappers from register_aten_ops.cpp, but register
them with c10 directly instead of JIT.

The `use_c10_dispatcher` setting in `native_functions.yaml` now has a new option 'with_codegenerated_unboxing_wrapper' which means we take the codegened unboxing wrapper from register_aten_ops.cpp and stuff it into c10. This new argument is the default, 'unboxed_only' is not the default anymore. For the ops that don't support boxed dispatch yet (i.e. ops taking TensorOptions arguments), we set them to 'unboxed_only' and they follow the old behavior of having register_aten_ops.cpp register the jit op.

This temporary hack will allow us to get mobile working much faster, but the plan is
to remove this hack again once all ops support the templated unboxing wrapper.

Differential Revision: [D18672405](https://our.internmc.facebook.com/intern/diff/D18672405/)

[ghstack-poisoned]
